src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ModuleBuilderImpl.cs

```diff
@@ -37,8 +37,10 @@ internal sealed class ModuleBuilderImpl : ModuleBuilder
         private bool _hasGlobalBeenCreated;
         private Type?[]? _coreTypes;
         private MetadataBuilder _pdbBuilder = new();
-        private static readonly Type[] s_coreTypes = { typeof(void), typeof(object), typeof(bool), typeof(char), typeof(sbyte), typeof(byte), typeof(short), typeof(ushort), typeof(int),
-                                                       typeof(uint), typeof(long), typeof(ulong), typeof(float), typeof(double), typeof(string), typeof(nint), typeof(nuint), typeof(TypedReference) };
+        private static readonly Type[] s_coreTypes = { typeof(void), typeof(object), typeof(bool), typeof(char), typeof(sbyte),
```

Comment:
Does the order here matter? In the sense does it need to match the order of the `CoreTypeId` enum? If so, I would add a comment.

Reply:
In both places.

Reply:
Right, order should match, added comments